src/Generator/Generators/CSharp/CSharpSources.cs

```diff
@@ -655,26 +657,16 @@ private void GenerateClassInternalsField(LayoutField field, Class @class)
             if (!Options.GenerateSequentialLayout || @class.IsUnion)
                 WriteLine($"[FieldOffset({field.Offset})]");
 
-            TypePrinter.PushMarshalKind(MarshalKind.NativeField);
-            var fieldTypePrinted = field.QualifiedType.CSharpType(TypePrinter);
-            TypePrinter.PopMarshalKind();
-
-            var fieldType = field.QualifiedType.Type.Desugar().IsAddress() ?
-                CSharpTypePrinter.IntPtrType : fieldTypePrinted.Type;
-
-            var fieldName = safeIdentifier;
-            if (!string.IsNullOrWhiteSpace(fieldTypePrinted.NameSuffix))
-                fieldName += fieldTypePrinted.NameSuffix;
-
+           
             var access = decl != null && !decl.IsGenerated ? "internal" : "public";
             if (field.Expression != null)
             {
                 var fieldValuePrinted = field.Expression.CSharpValue(ExpressionPrinter);
-                Write($"{access} {fieldType} {fieldName} = {fieldValuePrinted};");
+                Write($"{access} {retType.NameSuffix} {retType.Type} = {fieldValuePrinted};");
```

Comment:
I think you use the "suffix" as a prefix instead. In general, do not work with the suffix here, just add it to retType.Type in the type printer, and ensure it's put where it was before i.e. at the end.